server/sonar-server/src/main/java/org/sonar/server/branch/ws/ListAction.java

```diff
@@ -97,30 +97,34 @@ public void handle(Request request, Response response) throws Exception {
       userSession.checkComponentPermission(UserRole.USER, project);
       checkArgument(project.isEnabled() && PROJECT.equals(project.qualifier()), "Invalid project key");
 
-      Collection<BranchDto> branches = dbClient.branchDao().selectByComponent(dbSession, project);
+      Collection<BranchDto> branches = dbClient.branchDao().selectByComponent(dbSession, project).stream()
+        .filter(b -> b.getBranchType() == SHORT || b.getBranchType() == LONG)
```

Comment:
Maybe add a TODO to replace with condition that depends on key type, when it's ready.